bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -517,8 +547,22 @@ Identity SDFFeatures::ConstructSdfModelImpl(
           auto worldIdentity = modelIt->second->world;
           auto modelIdentity =
               this->GenerateIdentity(_modelOrWorldID, modelIt->second);
+
+          std::string modelScopedName;
+          math::Pose3d modelPose;
+          if (!getModelScopedName(_model, &_sdfModel,
+              _sdfModel.Name() + "::", modelScopedName, getModelScopedName))
+            this->GenerateInvalidId();
+          auto errors = _sdfModel.SemanticPose().Resolve(modelPose,
+              modelScopedName);
+          if (!errors.empty())
+            this->GenerateInvalidId();
+          auto modelToNestedModel = math::eigen3::convert(modelPose).inverse();
+
           return this->AddNestedModel(
-              _model->Name(), modelIdentity, worldIdentity, rootInertialToLink,
+              _model->Name(), modelIdentity, worldIdentity,
+              *modelToRootLink, rootInertialToLink,
```

Comment:
What do you think about using applying the extra transform from the root link to the nested model here so that we don't have to apply a special case for nested models in `KinemtacisFeatures`?

Reply:
ok I think I understood what you mean now. I refactored the code in [77e8a6e](https://github.com/gazebosim/gz-physics/pull/649/commits/77e8a6ef9d7cf675f79a31b16541ff81c93759d5). I was not able to do `structure.rootLink->SemanticPose()->Resolve` because I think `structure.rootLink` does not have a view of the whole SDF tree containing the nested model so I computed the transform manually.